src/executables/srpd_rotation.c

```diff
@@ -0,0 +1,398 @@
+            path = NULL;
+        }
+    }
+
+    return path;
+}
+
+int
+srpd_get_rot_count_cb(sr_session_ctx_t *session, uint32_t sub_id, const char *module_name, const char *path,
+        const char *request_xpath, uint32_t request_id, struct lyd_node **parent, void *private_data)
+{
+    int rc = 0;
+    const struct ly_ctx *ctx;
+    char value[20];
+    srpd_rotation_opts_t *opts = (srpd_rotation_opts_t *)private_data;
+
+    (void) sub_id;
+    (void) module_name;
+    (void) path;
+    (void) request_xpath;
+    (void) request_id;
+
+    sprintf(value, "%" PRIu64, ATOMIC_LOAD_RELAXED(opts->rotated_files_count));
+    ctx = sr_session_acquire_context(session);
+    if ((rc = lyd_new_path(*parent, ctx, "/sysrepo-plugind:sysrepo-plugind/notif-datastore/rotation/rotated-files-count",
+            value, 0, NULL)) != LY_SUCCESS) {
+        goto cleanup;
+    }
+
+cleanup:
+    sr_session_release_context(session);
+    return rc;
+}
+
+/**
+ * @brief Checks whether the format of the file name
+ * matches std format of notification file.
+ *
+ * @param[in] file_name File name to be checked.
+ * @param[out] file_time1 First derived time from the file name.
+ * @param[out] file_time2 Second derived time from the file name.
+ * @return 0 on success.
+ * @return 1 on failure.
+ */
+static int
+srpd_format_check(const char *file_name, time_t *file_time1, time_t *file_time2)
+{
+    char *x = NULL, *endptr = NULL;
+    time_t time1, time2;
+    char buf[7];
+
+    memset(buf, '\0', 7);
+
+    if (file_name == NULL) {
+        return EXIT_FAILURE;
+    }
+    x = strchr(file_name, '.');
+    if (x == NULL) {
+        return EXIT_FAILURE;
+    }
+    strncpy(buf, x + 1, 6);
+    if (strcmp(buf, "notif.")) {
+        return EXIT_FAILURE;
+    }
+    x = strchr(x + 1, '.');
+    time1 = strtoul(x + 1, &endptr, 10);
+    if (!strcmp(endptr, x + 1)) {
+        return EXIT_FAILURE;
+    }
+    x = strchr(x + 1, '-');
+    if (x == NULL) {
+        return EXIT_FAILURE;
+    }
+    time2 = strtoul(x + 1, &endptr, 10);
+    if (!strcmp(endptr, x + 1)) {
+        return EXIT_FAILURE;
+    }
+
+    if (file_time1 != NULL) {
+        *file_time1 = time1;
+    }
+    if (file_time2 != NULL) {
+        *file_time2 = time2;
+    }
+    return 0;
+}
+
+static void *
+srpd_rotation_loop(void *arg)
+{
+    int rc = 0;
+    DIR *d = NULL;
+    struct dirent *dir = NULL;
+    time_t current_time, file_time2 = 0;
+    srpd_rotation_opts_t *opts = (srpd_rotation_opts_t *)arg;
+    char *arg1 = NULL, *arg2 = NULL, *remove_str = NULL, *notif_dir_name = NULL;
+
+    notif_dir_name = srpd_get_notif_path();
+    if (!notif_dir_name) {
+        SRPLG_LOG_ERR("srpd_rotation", "Fatal error: Notif directory is NULL.");
+        goto cleanup;
+    }
+    if (srpd_mkpath((char *)ATOMIC_PTR_LOAD_RELAXED(opts->output_folder), 0777) == -1) {
+        SRPLG_LOG_ERR("srpd_rotation", "Fatal error: Archive directory could not be created");
```

Comment:
Redundant `Fatal error`, it is obvious from the message severity.

Reply:
Fixed